### ISSUE TITLE
Impala `CREATE TABLE AS SELECT` unparsable

### ISSUE THREAD

**Issue by mrebaker** [sqlfluff/sqlfluff#6451]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLfluff fails to parse `CREATE TABLE ... AS SELECT ...` queries, despite being supported by Impala - see [docs](https://impala.apache.org/docs/build/html/topics/impala_create_table.html).

### Expected Behaviour

This should be parsed without any failures.

### Observed Behaviour

```
== [stdin] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE foo AS SELECT col FROM bar'
WARNING: Parsing errors found and dialect is set to 'impala'. Have you configured your dialect correctly?
```

### How to reproduce

```sh
echo 'CREATE TABLE foo AS SELECT col FROM bar' | sqlfluff lint --dialect=impala -
```

### Dialect

Impala

### Version

sqlfluff: not yet a version - main @ 0e333c3
Python: 3.10.13

### Configuration

```
[sqlfluff]
dialect = impala
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
